### ISSUE TITLE
some doubts

### ISSUE THREAD

**Issue by liangshenglei** [RammusLeo/DPMesh#9]
Very excellent job, I have been studying your code recently and have some doubts.
1. Noisy 2D heatmap: I found that this heatmap was not obtained by the off the shelf 2D key point detector, but by masking 10% of the joints. I don't know if my understanding is correct.
2. Visible joints 2D: From the code, it appears to be a 2D joint from GT. I don't know if my understanding is correct.
3. Noisy Key point Reasoning: I didn't find this part of the code in the model, I only found the VPDTacherModel, but it doesn't seem to work.
I hope you can answer my doubts.
May everything be well!

**Comment by RammusLeo** [comment]
Sorry for late reply! 
1. During training, we use the noisy gt 2d keypoints for training because it is a huge work to obtain the 2d keypoints for such amount of training data. But during inference we use the detected keypoints. You can further check the test dataloader.
2. Same as above, we use predicted keypoints as the inputs. You can also run out your results for in-the-wild images based on detected keypoints and our released models. Our model works well on various environments.
3. You can try to train the teacher using the GT labels the same way as training student. 
Best wishes!

**Comment by ouusan** [comment]
@RammusLeo @liangshenglei   Hello! 
I have some question about Noisy Key point Reasoning：the current training code does not include any part for distillation learning using a teacher model, right? for using distillation learning, we should first pre-train a teacher model. Then, we define a student model and use the teacher model's outputs {"feat_3d": dense_feat, "feat_2d": img} to compute the loss between the teacher's outputs and the student model's outputs. Specifically, we supervise the feat_2d and feat_3d values from the teacher model with the outputs of the student model. Is my understanding correct, or did I miss something? and last question: What kind of network should the student model use? Are there any key points I should pay attention to?? Thank you in advance!
